### PR TITLE
feat: Bump IPFS/IPLD deps

### PR DESCRIPTION
**Pull request by ukstv**
And Jest.

### CODE REVIEW

File: packages/core/src/__tests__/__snapshots__/caip10-link.test.ts.snap [hunk 1]
```diff
@@ -297,61 +112,24 @@ Object {
...
-          14,
-          178,
-          67,
-          157,
-        ],
-        "version": 1,
+  "log": [
+    {
+      "cid": {
+        "/": "bafyreig4iuf5hwilecv5tnj7ye6bypuqhzsyylzbhgnp5a5hkqyq5msdtu",
```
Comment (oed) [inline packages/core/src/__tests__/__snapshots__/caip10-link.test.ts.snap:121]: This is such an improvement to the CID lib!

File: packages/core/src/pubsub/__tests__/pubsub.test.ts [hunk 2]
```diff
@@ -17,27 +17,33 @@ const diagnosticsLogger = loggerProvider.getDiagnosticsLogger()
...
     data: random.randomBytes(32),
-    topicIDs: [TOPIC],
-    seqno: random.randomBytes(10),
+    sequenceNumber: BigInt(random.randomUint32()),
+    signature: random.randomBytes(32),
+    key: random.randomBytes(32),
   }
 })
-const LATE_MESSAGE_AFTER = 1000
+const LATE_MESSAGE_AFTER = 200
```
Comment (oed) [inline packages/core/src/pubsub/__tests__/pubsub.test.ts:46]: Why was this constant changed?

**Review by oed — CHANGES_REQUESTED** [review]
Looks good. Just one request to keep the code for swapping ipfs backend.

File: packages/core/src/pubsub/incoming-channel.ts [hunk 3]
```diff
@@ -98,10 +88,15 @@ export class IncomingChannel extends Observable<IPFSPubsubMessage> {
  */
 export function filterExternal(
   ownPeerId$: Observable<string>
-): MonoTypeOperatorFunction<IPFSPubsubMessage> {
+): MonoTypeOperatorFunction<SignedMessage> {
   return pipe(
-    concatMap((data: IPFSPubsubMessage) => {
-      return ownPeerId$.pipe(map((peerId) => ({ isOuter: data.from !== peerId, entry: data })))
+    filter((message) => message.type === 'signed'),
```
[the remark this answers is not in the document]
Comment (ukstv) [reply packages/core/src/pubsub/incoming-channel.ts:93]: IPFS type system suggests, there could be "unsigned". Added the filter to be on the safe side.

File: packages/ipfs-daemon/src/create-ipfs.ts [hunk 4]
```diff
@@ -106,17 +107,15 @@ const createInstanceByType = {
  * @param overrideConfig - IFPS config for override
  */
 export async function createIPFS(
-  overrideConfig: Partial<Options> = {},
+  overrideConfig: Partial<Ctl.IPFSOptions> = {},
   disposable = true
 ): Promise<IpfsApi> {
-  const flavor = process.env.IPFS_FLAVOR || 'go'
-
```
[the remark this answers is not in the document]
Comment (ukstv) [reply packages/ipfs-daemon/src/create-ipfs.ts:113]: OHH, okay.

**Review by oed — APPROVED** [review]

File: packages/ipfs-daemon/src/ipfs-daemon.ts [hunk 5]
```diff
@@ -91,17 +90,8 @@ export class IpfsDaemon {
...
         repo: configuration.ipfsPath,
-        ipld: { codecs: [dagJose] },
         libp2p: {
-          config: {
-            dht: {
-              enabled: false,
-              clientMode: !configuration.ipfsDhtServerMode,
-            },
-            pubsub: {
-              enabled: configuration.ipfsEnablePubsub,
```
Comment (smrz2001) [inline packages/ipfs-daemon/src/ipfs-daemon.ts:102]: Does this have some impact?

Comment (oed) [reply packages/ipfs-daemon/src/ipfs-daemon.ts:102]: I believe this config is only for js-ipfs if I'm not mistaken, and we disabled the js-ipfs tests

Comment (ukstv) [reply packages/ipfs-daemon/src/ipfs-daemon.ts:102]: Good catch. I do not know, to be honest. Let me get it back indeed. I finally figured the types.

**Comment by oed** [comment]
✅ ✅ ✅

**Review by smrz2001 — APPROVED** [review]
:rocket:

**Review by stbrody — APPROVED** [review]
🚀

**Pull request merged**